core/domain/role_services.py

```diff
@@ -131,3 +142,71 @@ def get_all_actions(role):
         role_actions.extend(get_all_actions(parent_role))
 
     return list(set(role_actions))
+
+
+def get_updatable_roles():
+    """Returns roles to which a user's role can be updated by admin interface.
+
+    Returns:
+        dict. A dict of all roles that a user can be given by admin.
+            key -> roles in format as stored in backed.
+            value -> roles in format to show users.
+    """
+    updatable_roles = {}
+    for role in PARENT_ROLES:
+        if role not in [feconf.ROLE_GUEST]:
+            updatable_roles[role] = get_human_readable_role(role)
+    return updatable_roles
+
+
+def get_viewable_roles():
+    """Returns roles by which users can be searched in admin interface.
+
+    Returns:
+        dict. A dict of all roles by which user's can be searched.
+            key -> roles in format as stored in backend.
```

Comment:
Actually, I like the format @thatbrod is introducing here: https://github.com/oppia/oppia/pull/3519/files#diff-9d1f2f6e654f24f2e509b57e240d6f07R501

So maybe we can do:

    dict(str : str). Each key of this dict is a role ID, and the corresponding value is its human-readable name to display in the UI.

or similar.

Btw, what are we calling the backend role strings? Role ID or role string or ...? I like "role ID" since it mirrors "interaction ID" etc., but I'm not sure we've established that terminology yet. "Role string" seems a little vague.